scenario/clickactionbutton.go

```diff
@@ -0,0 +1,643 @@
+				func(fieldName string) (*enigma.Field, error) {
+					return doc.GetField(ctx, fieldName)
+				},
+			)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			if values := toFieldValues(buttonAction.Value); len(values) != 0 {
+				_, err = field.SelectValues(ctx, values, false /*toggleMode*/, buttonAction.SoftLock)
+			}
+			return errors.Wrapf(err, "Could not select values in field<%s>", buttonAction.Field)
+		})
+
+	case selectValuesMatchingSearchCriteria:
+		return sendReq(func(ctx context.Context) error {
+			field, err := uplink.FieldCache.LookupWithFallback(buttonAction.Field,
+				func(fieldName string) (*enigma.Field, error) {
+					return doc.GetField(ctx, fieldName)
+				},
+			)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			_, err = field.Select(ctx, buttonAction.Value /*match*/, buttonAction.SoftLock, 0 /*excludedValuesMode*/)
+			return errors.Wrapf(err, "Selection failed in field<%s> searchcritera<%s>", buttonAction.Field, buttonAction.Value)
+		})
+
+	case selectAlternatives:
+		return sendReq(func(ctx context.Context) error {
+			field, err := uplink.FieldCache.LookupWithFallback(buttonAction.Field,
+				func(fieldName string) (*enigma.Field, error) {
+					return doc.GetField(ctx, fieldName)
+				},
+			)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			_, err = field.SelectAlternative(ctx, buttonAction.SoftLock)
+			return err
+		})
+
+	case selectExcluded:
+		return sendReq(func(ctx context.Context) error {
+			field, err := uplink.FieldCache.LookupWithFallback(buttonAction.Field,
+				func(fieldName string) (*enigma.Field, error) {
+					return doc.GetField(ctx, fieldName)
+				},
+			)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			_, err = field.SelectExcluded(ctx, buttonAction.SoftLock)
+			return err
+		})
+
+	case selectPossibleValuesInField:
+		return sendReq(func(ctx context.Context) error {
+			field, err := uplink.FieldCache.LookupWithFallback(buttonAction.Field,
+				func(fieldName string) (*enigma.Field, error) {
+					return doc.GetField(ctx, fieldName)
+				},
+			)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			_, err = field.SelectPossible(ctx, buttonAction.SoftLock)
+			return errors.Wrapf(err, "Could not select possible in field<%s>", buttonAction.Field)
+		})
+
+	case toggleFieldSelection:
+		return sendReq(func(ctx context.Context) error {
+			field, err := uplink.FieldCache.LookupWithFallback(buttonAction.Field,
+				func(fieldName string) (*enigma.Field, error) {
+					return doc.GetField(ctx, fieldName)
+				},
+			)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			_, err = field.ToggleSelect(ctx, buttonAction.Value /*match*/, buttonAction.SoftLock, 0 /*excludedValuesMode*/)
+			return errors.Wrapf(err, "Could not select possible in field<%s>", buttonAction.Field)
+		})
+
+	case lockAllSelections:
+		return sendReq(func(ctx context.Context) error {
+			return doc.LockAll(ctx, "" /*stateName*/)
```

Comment:
Might be interesting to keep track of this internally so that we can send an appropriate warning if the user later does something which is not allowed under the current state.

Reply:
Very true. Will take a look at where.